### PR TITLE
Fix underspecified insert tests

### PR DESCRIPTION
**Pull request by max-hoffman**
No inserts were not being run with prepareds. A lot of them were broken, but should be running in GMS now at least.

Edit: These pass Dolt also.

### CODE REVIEW

File: enginetest/enginetests.go [hunk 1]
```diff
@@ -6492,7 +6492,19 @@ func runQueryPreparedWithCtx(
 	require := require.New(t)
 	parsed, err := parse.Parse(ctx, q)
 
-	if _, ok := parsed.(sql.Databaser); ok {
+	_, isInsert := parsed.(*plan.InsertInto)
+	_, isDatabaser := parsed.(sql.Databaser)
+	if bl, ok := parsed.(*plan.Block); ok {
```
Comment (zachmu) [inline enginetest/enginetests.go:6497]: When is an insert ever inside a block node? What's going on here?

**Review by zachmu — APPROVED** [review]
A little mysterious, could use some explanatory comments

File: enginetest/enginetests.go (hunk 1 above)
Comment (max-hoffman) [reply enginetest/enginetests.go:6497]: this one also confuses me, daylon added Block last week for foreign keys

Comment (max-hoffman) [reply enginetest/enginetests.go:6497]: looks like `*sqlparser.MultiAlterDDL` is now a thing, Block wraps arbitrarily nested statements?

File: sql/analyzer/inserts.go [hunk 2]
```diff
@@ -108,6 +108,29 @@ func resolveInsertRows(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope,
...
+// resolvePreparedInsert applies post-optimization
+// rules to Insert.Source for prepared statements.
+func resolvePreparedInsert(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
+	return transform.Node(n, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
+		ins, ok := n.(*plan.InsertInto)
+		if !ok {
+			return n, transform.SameTree, nil
+		}
+
+		if _, ok := ins.Source.(*plan.TriggerExecutor); ok {
```
[the remark this answers is not in the document]
Comment (max-hoffman) [reply sql/analyzer/inserts.go:120]: i'll copy the comment from above

**Pull request merged**